### ISSUE TITLE
Links with brackets in them parse incorrectly

### ISSUE THREAD

**Issue by tmladek** [johannhof/markdown.rs#34]
This may not be a `markdown.rs` issue than a markdown issue, but even on GitHub, a link such as this:

`[John Gall](https://en.wikipedia.org/wiki/John_Gall_(author))`

parses as: [John Gall](https://en.wikipedia.org/wiki/John_Gall_(author))

Whereas `markdown.rs` parses it as `<a href="https://en.wikipedia.org/wiki/John_Gall_(author">John Gall</a>)`, resulting in a hanging bracket and a broken link.

I understand that it may be difficult to counter given that most of the parsing is done via regex, but I think it's worth trying to fix. Thanks in advance!

**Comment by johannhof** [comment]
Yeah that is actually fixed in master I think, but I haven't made a new release yet. I'll do that tonight
